### PR TITLE
[masonry-layout] Added definitions for hiddenStyle and visibleStyle

### PR DESCRIPTION
**Pull request by Serraniel**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://codepen.io/desandro/pen/bqvLaZ/ shows how to use those two options. This example is linked in the [docs](https://masonry.desandro.com/extras.html). Additionally you can see usage in code when they set the defaults: https://github.com/desandro/masonry/blob/master/dist/masonry.pkgd.js#L1438-L1444
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/masonry-layout/index.d.ts [hunk 1]
```diff
@@ -40,6 +40,11 @@ declare class Masonry {
 }
 
 declare namespace Masonry {
+    interface hiddenOrVisibleStyleOption {
```
Comment (peterblazejewicz) [inline types/masonry-layout/index.d.ts:43]:
```suggestion
    interface HiddenOrVisibleStyleOption {
```
https://palantir.github.io/tslint/rules/class-name/
I'm not sure if it's enforced here, but pretty much used everhwere

File: types/masonry-layout/index.d.ts [hunk 2]
```diff
@@ -51,10 +56,13 @@ declare namespace Masonry {
         originLeft?: boolean;
         originTop?: boolean;
         horizontalOrder?: boolean;
+        hiddenStyle?: hiddenOrVisibleStyleOption;
```
Comment (peterblazejewicz) [inline types/masonry-layout/index.d.ts:59]:
```suggestion
        hiddenStyle?: HiddenOrVisibleStyleOption;
```
ditto

File: types/masonry-layout/index.d.ts [hunk 3]
```diff
@@ -51,10 +56,13 @@ declare namespace Masonry {
...
+        visibleStyle?: hiddenOrVisibleStyleOption;
```
Comment (peterblazejewicz) [inline types/masonry-layout/index.d.ts:60]:
```suggestion
        visibleStyle?: HiddenOrVisibleStyleOption;
```

File: types/masonry-layout/index.d.ts [hunk 4]
```diff
@@ -51,10 +56,13 @@ declare namespace Masonry {
...
 
         // setup
         containerStyle?: {};
         transitionDuration?: any;
+        stagger?: string | number;
```
Comment (peterblazejewicz) [inline types/masonry-layout/index.d.ts:65]: just question, if you want to accept nulls:
https://github.com/desandro/masonry/blob/master/dist/masonry.pkgd.js#L1684

**Review by peterblazejewicz — COMMENTED** [review]
@Serraniel pls review my comments, thx!

File: types/masonry-layout/index.d.ts (hunk 4 above)
Comment (Serraniel) [reply types/masonry-layout/index.d.ts:65]: Just noticed I forgot to add the docs link for this:
https://masonry.desandro.com/options.html#stagger

The above link describes this option. There also is the following sentence on top of the page:
> All options are optional, but columnWidth and itemSelector are recommended.

**Review by peterblazejewicz — APPROVED** [review]
LGMT!

**Comment by Serraniel** [comment]
Ready to merge

**Pull request merged**